grafana/resource_machine_learning_outlier_detector.go

```diff
@@ -0,0 +1,260 @@
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The ID of the outlier detector.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				Description: "The name of the outlier detector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"metric": {
+				Description: "The metric used to query the outlier detector results.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "A description of the outlier detector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"datasource_id": {
+				Description: "The id of the datasource to query.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			"datasource_uid": {
+				Description: "The uid of the datasource to query.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"datasource_type": {
+				Description:  "The type of datasource being queried. Currently allowed values are prometheus, graphite, loki, postgres, and datadog.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"prometheus", "graphite", "loki", "postgres", "datadog"}, false),
+			},
+			"query_params": {
+				Description: "An object representing the query params to query Grafana with.",
+				Type:        schema.TypeMap,
+				Required:    true,
+			},
+			"interval": {
+				Description: "The data interval in seconds to monitor.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     300,
+			},
+			"algorithm": {
+				Description: "The algorithm to use and its configuration. See https://grafana.com/docs/grafana-cloud/machine-learning/outlier-detection/ for details.",
+				Type:        schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description:  "The name of the algorithm to use ('mad' or 'dbscan').",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"mad", "dbscan"}, false),
+						},
+						"sensitivity": {
+							Description:  "Specify the sensitivity of the detector (in range [0,1]).",
+							Type:         schema.TypeFloat,
+							Required:     true,
+							ValidateFunc: validation.FloatBetween(0, 1.0),
+						},
+						"config": {
+							Description: "For DBSCAN only, specify the configuration map",
+							Type:        schema.TypeSet,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"epsilon": {
+										Description:  "Specify the epsilon parameter (positive float)",
+										Type:         schema.TypeFloat,
+										Required:     true,
+										ValidateFunc: validation.FloatAtLeast(0),
+									},
+								},
+							},
+							Optional: true,
+						},
+					},
+				},
+				Required: true,
```

Comment:
Can you move these two attributes before their sibling `Elem` attribute? Otherwise it is hard to know which resource are they being applied to.

Reply:
Done